### PR TITLE
Translate

### PR DESCRIPTION
**Pull request by dallen68**

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,60 @@
...
+        });
+
+        if (eventList.isEmpty()) {
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+        
+        List<TimeRange> availableTimes = new ArrayList();
+        if (eventList.get(0).getWhen().start() - TimeRange.START_OF_DAY > request.getDuration()) {
+            availableTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY,eventList.get(0).getWhen().start(),false));
+        }
```
Comment (LiRq95) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:50]: It can be somewhat hard to follow the logic around this part. But so far I see no glaring issues? But I would recommend adding a lot more comments in this class so if someone else like Derrill looks it over it'll be quicker for him to see what's going on (Incidentally, a scheduler-related question happens to be one of my go-to interview questions, so I already had some familiarity with this solution).

Comment (dallen68) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:103]: I added javadocs and moved some code to helper functions. I wasn't sure how to do this without adding my other commit.

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 2]
```diff
@@ -270,5 +272,139 @@ public void notEnoughRoom() {
...
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noMandatoryAttendeesWithGaps() {
+
```
Comment (LiRq95) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:366]: Remove empty newlines (did you mean to include a comment here like the other tests?)

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 3]
```diff
@@ -270,5 +272,139 @@ public void notEnoughRoom() {
...
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noMandatoryAttendeesWithNoGaps() {
+
```
Comment (LiRq95) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:390]: Remove empty newlines
